includes/main.inc.php

```diff
@@ -156,14 +156,17 @@ function stripslashes_deep($value) {
 	}
 }
 
-if (isset($_GET['toggle_view'])) {
-	if (isset($_SESSION[$settings['session_prefix'].'usersettings']) && $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] == 0) {
-		$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 1;
-		setcookie($settings['session_prefix'].'usersettings', '1.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'], TIMESTAMP + (3600 * 24 *$settings['cookie_validity_days']));
-	} elseif (isset($_SESSION[$settings['session_prefix'].'usersettings']) && $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] == 1) {
-    $_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = 0;
-    setcookie($settings['session_prefix'].'usersettings','0.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'],TIMESTAMP+(3600*24*$settings['cookie_validity_days']));
+if (isset($_GET['toggle_view']) or isset($_GET['main_view'])) {
```

Comment:
Why we need a new parameter? Maybe, I'm wrong but I believe, we can keep it more compact, i.e.:

```
if (isset($_SESSION[$settings['session_prefix'].'usersettings']) && isset($_GET['toggle_view']) && in_array($_GET['toggle_view'], array(0, 1))) {
	$_SESSION[$settings['session_prefix'].'usersettings']['user_view'] = intval($_GET['toggle_view']);
	setcookie($settings['session_prefix'].'usersettings', intval($_GET['toggle_view']).'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_order'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['sidebar'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['fold_threads'].'.'.$_SESSION[$settings['session_prefix'].'usersettings']['thread_display'],TIMESTAMP+(3600*24*$settings['cookie_validity_days']));
}
```

Reply:
I decided to keep `toogle_view` as the simple switch, that it is at the moment. If it's set, the view will be switched from one to the other. Because of the idea to keep it in it's actual mode of operation and for the special use case of (external) links to a defined view I introduced the second parameter.

It should IMHO be an independent way of defining the whished view. What's with existing links to forums, that works with `toggle_view` in it's actual mode of operations? I assume, that this happens not very often, but i.e. [Pedro does it](http://mylittleforum.net/forum/index.php?id=10170) from an external program. With a further parameter no interference with existing behaviour should arise.

It is also possible to change the mode of operation of the parameter `toggle_view` (what I wanted to prevent with my approach). In that case we need to search for every place, where the URL-string get's defined and change it with the actual status in mind.

... And Pedro has to check all his current existing links. hehe :-)